### PR TITLE
python3Packages.pymodes: init at 2.11

### PR DESCRIPTION
**Pull request by snicket2100**
###### Description of changes

https://github.com/junzis/pyModeS - Python ADS-B/Mode-S Decoder

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pymodes/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchPypi, buildPythonPackage, numpy, pyzmq }:
+
+buildPythonPackage rec {
+  pname = "pyModeS";
```
Comment (fabaff) [inline pkgs/development/python-modules/pymodes/default.nix:4]:
```suggestion
  pname = "pymodes";
```

Naming should follow [PEP-503](https://nixos.org/manual/nixpkgs/stable/#contributing).

File: pkgs/development/python-modules/pymodes/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "2.11";
+
```
Comment (fabaff) [inline pkgs/development/python-modules/pymodes/default.nix:6]: The source only says Python 3 but I suggest to limit it to supported Python releases. 

```suggestion
  format = "setuptools";

  disabled = pythonOlder "3.7";
```

File: pkgs/development/python-modules/pymodes/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1r85scs99c87mqkjalw1wxzscnibyyvzdgyj6q8l0f8r5f2864c2";
+  };
+
+  propagatedBuildInputs = [ numpy pyzmq ];
+
+  # No tests included
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/pymodes/default.nix:15]: The upstream repo contains a folder [`tests`](https://github.com/junzis/pyModeS/tree/master/tests). Please use `fetchFromGitHub` to get the complete source  and run the tests.

File: pkgs/development/python-modules/pymodes/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  pythonImportsCheck = [ "pyModeS" ];
+
+  meta = with lib; {
+    description = "Python Mode-S and ADS-B Decoder";
+    homepage = "https://github.com/junzis/pyModeS";
+    license = licenses.gpl3;
```
Comment (fabaff) [inline pkgs/development/python-modules/pymodes/default.nix:22]: One file's header says `GPLv3+`.

```suggestion
    license = licenses.gpl3Plus;
```

File: pkgs/development/python-modules/pymodes/default.nix (hunk 1 above)
Comment (snicket2100) [reply pkgs/development/python-modules/pymodes/default.nix]: Fixed.

File: pkgs/development/python-modules/pymodes/default.nix (hunk 3 above)
Comment (snicket2100) [reply pkgs/development/python-modules/pymodes/default.nix]: Done.

File: pkgs/development/python-modules/pymodes/default.nix (hunk 4 above)
Comment (snicket2100) [reply pkgs/development/python-modules/pymodes/default.nix]: Fixed.

File: pkgs/development/python-modules/pymodes/default.nix (hunk 2 above)
Comment (snicket2100) [reply pkgs/development/python-modules/pymodes/default.nix:13]: Done.

**Comment by snicket2100** [comment]
Thanks for the review @fabaff, I have applied all of your comments.

File: pkgs/development/python-modules/pymodes/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{ lib
+, stdenv
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pymodes/default.nix:2]: Please format with the new formatter.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by fabaff** [comment]
Move to a new PR https://github.com/NixOS/nixpkgs/pull/359536

**Pull request closed without merging**